mfr/extensions/tabular/libs/xlrd_tools.py

```diff
@@ -17,29 +17,30 @@ def xlsx_xlrd(fp):
 
     wb = xlrd.open_workbook(fp.name)
 
-    # Currently only displays the first sheet if there are more than one.
-    sheet = wb.sheets()[0]
+    sheets = {}
 
-    if sheet.ncols > max_size or sheet.nrows > max_size:
-        raise TableTooBigException("Table is too large to render.")
+    for sheet in wb.sheets():
+        if sheet.ncols > max_size or sheet.nrows > max_size:
+            raise TableTooBigException("Table is too large to render.")
 
-    if sheet.ncols < 1 or sheet.nrows < 1:
-        raise EmptyTableException("Table is empty or corrupt.")
+        if sheet.ncols < 1 or sheet.nrows < 1:
```

Comment:
This exception needs to only be shown if no data is available for **ALL** sheets. In this current configuration having a single blank sheet which is **VERY** common in excel causes this to be raised thus the user never gets to see any workbook data.